subxt/src/client/lightclient/rpc.rs

```diff
@@ -0,0 +1,215 @@
+// Copyright 2019-2023 Parity Technologies (UK) Ltd.
+// This file is dual-licensed as Apache-2.0 or GPL-3.0.
+// see LICENSE for license details.
+
+use super::{
+    background::{BackgroundTask, FromSubxt, MethodResponse},
+    LightClientError,
+};
+use crate::{
+    error::{Error, RpcError},
+    rpc::{RpcClientT, RpcFuture, RpcSubscription},
+};
+use futures::{stream::StreamExt, Stream};
+use serde_json::value::RawValue;
+use smoldot_light::{platform::async_std::AsyncStdTcpWebSocket as Platform, ChainId};
+use std::pin::Pin;
+use tokio::sync::{mpsc, mpsc::error::SendError, oneshot};
+use tokio_stream::wrappers::ReceiverStream;
+
+pub const LOG_TARGET: &str = "light-client";
+
+/// The light-client RPC implementation that is used to connect with the chain.
+#[derive(Clone)]
+pub struct LightClientRpc {
+    /// Communicate with the backend task that multiplexes the responses
+    /// back to the frontend.
+    to_backend: mpsc::Sender<FromSubxt>,
+}
+
+impl LightClientRpc {
+    /// Constructs a new [`LightClientRpc`], providing the chain specification.
+    ///
+    /// The chain specification can be downloaded from a trusted network via
+    /// the `sync_state_genSyncSpec` RPC method. This parameter expects the
+    /// chain spec in text format (ie not in hex-encoded scale-encoded as RPC methods
+    /// will provide).
+    ///
+    /// ## Panics
+    ///
+    /// Panics if being called outside of `tokio` runtime context.
+    pub fn new(
+        config: smoldot_light::AddChainConfig<'_, (), impl Iterator<Item = ChainId>>,
+    ) -> Result<LightClientRpc, Error> {
+        tracing::trace!(target: LOG_TARGET, "Create light client");
+
+        let mut client = smoldot_light::Client::new(Platform::new(
+            env!("CARGO_PKG_NAME").into(),
+            env!("CARGO_PKG_VERSION").into(),
+        ));
+
+        let smoldot_light::AddChainSuccess {
+            chain_id,
+            json_rpc_responses,
+        } = client
+            .add_chain(config)
+            .map_err(|err| LightClientError::AddChainError(err.to_string()))?;
+
+        let (to_backend, backend) = mpsc::channel(128);
```

Comment:
could be good make this configurable at some point, just a warning that backpressure deadlocks are super hard to detect/debug.

Initially I would make these unbounded but your call :)

Reply:
Yep, I trust your intuition, have made these channels unbounded :D